lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -277,7 +277,10 @@ def add_test_targets
                 # requires frameworks. For tests we always use the test target name as the product name
                 # irrelevant to whether we use frameworks or not.
                 configuration.build_settings['PRODUCT_NAME'] = name
-                configuration.build_settings['PRODUCT_MODULE_NAME'] = name
+                # Use xcode default product module name, which is $(PRODUCT_NAME:c99extidentifier)
+                # this gives us always valid name that is distinct from the parent spec module name
+                # which allow tests to use either import or @testable import to access the parent framework
+                configuration.build_settings.delete('PRODUCT_MODULE_NAME')
```

Comment:
I dont think this will work, since we override `PRODUCT_MODULE_NAME` elsewhere

Reply:
we do but that happens earlier than this.

Reply:
looks like you right, as value gets inherited from parent spec and breaks compilation:
https://travis-ci.org/paulz/ImageCoordinateSpace/builds/354136985#L2284
```
The following build commands failed:
	CompileSwiftSources normal i386 com.apple.xcode.tools.swift.compiler
(1 failure)
 -> ImageCoordinateSpace (1.0.5)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [iOS] xcodebuild:  <unknown>:0: error: module name "ImageCoordinateSpace-Unit-Tests" is not a valid identifier
```

when cleared the default Xcode value works, here is branch demonstrating the successful `pod lib lint`: https://github.com/paulz/ImageCoordinateSpace/tree/test-spec on travis: https://travis-ci.org/paulz/ImageCoordinateSpace/builds/354903900